### ISSUE TITLE
Opencage lookup broken

### ISSUE THREAD

**Issue by nyejon** [DenisCarriere/geocoder#292]
Hi, 

The Opencage lookup seems to be broken again. I get the following error:

```
 File "..\lib\site-packages\geocoder\base.py", line 120, in _parse_json_with_fieldnames
    value = getattr(self, key)
  File "..\lib\site-packages\geocoder\opencage.py", line 359, in bbox
    south = self.raw['bounds']['southwest'].get('lat')
```

I can solve it by putting a try, except around the bit:

```
            try:
                value = getattr(self, key)
                if value:
                    self.json[key] = value
            except Exception as e:
                print(e)
```

but I'm not sure that that is the best solution... :)

I'm not sure how the new multiple search results have changed things and if it will be a general error, or only an error with opencage.

**Comment by nyejon** [comment]
Also another question, before if I did the following:
```
opencage = geocoder.opencage(address, key=settings.OPENCAGE_API_KEY, countrycode=country, no_annotations=1)
print(opencage.latlng, opencage.address)
```
It would just print None if there wasn't a result. Now it gives an error and I have to first do:

 ```
opencage = geocoder.opencage(address, key=settings.OPENCAGE_API_KEY, language='de', no_annotations=1)
if opencage.status == 'OK':
    print(opencage.latlng, opencage.address)
    return (opencage.lat, opencage.lng)

else:
    return (None, None)

```

is this the new intended behavior?

**Comment by ebreton** [comment]
Hi @nyejon ,

> The Opencage lookup seems to be broken again. I get the following error

Could you please provide the initial call you made (to allow me to reproduce) and the full stack (I can't see enough in your first message to diagnostic)

```
>>> from geocoder import opencage
>>> g = opencage('1552 Payette dr., Ottawa')
>>> g.bbox
{'northeast': [-75.4972762, 45.4701908], 'southwest': [-75.4974667, 45.4700396]}
```

> is this the new intended behavior?

nope. We wanted to be backward compatible. And this problem is fore sure a general one from base.py line 597.

This one is straightforward to fix (return None instead of raising an exception). 
Thanks for pointing it out !

I am therefore waiting on your feedback for the first one before making a PR to correct all of it.

Cheers,
Manu

**Comment by nyejon** [comment]
Hi,

I think it is just that there is no 'bounds' in many of the opencage json responses.

I am not even trying to use the bbox attribute, I think it is something to do with the:

```
        for key in dir(self):
            if not key.startswith('_') and key not in self._TO_EXCLUDE:
                self.fieldnames.append(key)
                value = getattr(self, key)
                if value:
                    self.json[key] = value
```

One of the lookups that provides the error is: 'AirportClinic M - MediCare Flughafen München Medizinisches Zentrum'. It returns a value, but it is the wrong location (it doesn't really find it).

Some of the json responses returns the 'bounds' if it is an exact lookup and then there is no error.

**Comment by ebreton** [comment]
Should be fixed with PR #293.

Could you check that it solves all your issues ? (with the branch fix_opencage)

**Comment by nyejon** [comment]
Thanks for the effort. It unfortunately still has an error:
With lookup again of:
'AirportClinic M - MediCare Flughafen München Medizinisches Zentrum'

```
Traceback (most recent call last):
  File "<console>", line 1, in <module>

  File "..\utils.py", line 11, in get_coordinates
    opencage = geocoder.opencage(address, key=settings.OPENCAGE_API_KEY, countrycode=country, language='de', no_annotations=1)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\api.py", line 367, in opencage
    return get(location, provider='opencage', **kwargs)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\api.py", line 175, in get
    return options[provider][method](location, **kwargs)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\base.py", line 407, in __init__
    self._initialize()
  File "..s\geocoder-1.32.1-py3.6.egg\geocoder\base.py", line 462, in _initialize
    self._parse_results(json_response)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\base.py", line 522, in _parse_results
    self.add(self.one_result(json_dict))
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\opencage.py", line 23, in __init__
    super(OpenCageResult, self).__init__(json_content)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\base.py", line 65, in __init__
    self._parse_json_with_fieldnames()
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\base.py", line 122, in _parse_json_with_fie
    value = getattr(self, key)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\opencage.py", line 372, in bbox
    return BBox.factory([south, west, north, east]).as_dict
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\location.py", line 163, in factory
    return cls(bbox=arg)
  File "..\geocoder-1.32.1-py3.6.egg\geocoder\location.py", line 175, in __init__
    self.west, self.south, self.east, self.north = map(float, bbox)
TypeError: float() argument must be a string or a number, not 'NoneType'
```

**Comment by ebreton** [comment]
I should have added a test case for this call ... 

it now passes on my side. Could you check again ?

**Comment by ebreton** [comment]
Merged into master by Denis 👍

**Comment by nyejon** [comment]
Thanks, tested this morning and all seems to be working again!
